### PR TITLE
Remove properly tracked config file from .gitignore & add support for skipping of link-checking

### PR DESCRIPTION
**Pull request by marxin**
The file is part of the git history and is a configuration file.

Fixes: #2018

### CODE REVIEW

**Comment by camelid** [comment]
I think the reason this is ignored is that we often want to change `book.toml` locally to turn off link-checking. I'm not sure it makes sense to remove it from the gitignore unless there's another way to disable link-checking locally.

**Comment by marxin** [comment]
Sure, that makes sense and I've just added the support. What do you think about it now?

**Comment by marxin** [comment]
@camelid May I please ping this?

**Comment by marxin** [comment]
@camelid: A gentle ping.

File: ci/linkcheck.sh [hunk 1]
```diff
@@ -3,10 +3,17 @@
...
 
+LINKCHECK_BINARY=mdbook-linkcheck2
+
 set_github_token() {
   jq '.config.output.linkcheck."http-headers"."github\\.com" = ["Authorization: Bearer $GITHUB_TOKEN"]'
 }
 
+if [ ! -z "$SKIP_LINKCHECK" ] ; then
+  echo "Skipping link check."
+  exec $LINKCHECK_BINARY -f ""
```
Comment (camelid) [inline ci/linkcheck.sh:14]: Shouldn't this be doing `exit 0` instead? I'm not sure why it's still running the linkcheck binary.

**Review by camelid — CHANGES_REQUESTED** [review]
Thanks! This is a much better approach than our old approach of just commenting out the config. Just left a comment about something that seems wrong.

File: ci/linkcheck.sh (hunk 1 above)
Comment (marxin) [reply ci/linkcheck.sh]: Yep, adjusted.

**Review by camelid — APPROVED** [review]

**Pull request merged**